packages/protocol/test/governance/network/blockchainparams.ts

```diff
@@ -114,30 +125,144 @@ contract('BlockchainParameters', (accounts: string[]) => {
+
+      // returns old value
+      assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), 20)
+    })
+  })
+
+  describe('#setUptimeLookbackWindow()', () => {
+    const lookbackWindow = 20
+
+    it('if set should set the variable for the next epoch', async () => {
+      // enter steady state
+      await blockchainParameters.setUptimeLookbackWindow(24)
+      await mineBlocks(EPOCH, web3)
+
+      await blockchainParameters.setUptimeLookbackWindow(40)
+
+      // still old value
+      assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), 24)
+
+      // wait an epoch to find new value
+      await mineBlocks(EPOCH, web3)
+      assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), 40)
+    })
+
+    it('if unset should set the variable for the current epoch', async () => {
+      await blockchainParameters.setUptimeLookbackWindow(24)
+      assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), 24)
+    })
+    it('if unset, second set should set the variable for the next epoch', async () => {
+      await blockchainParameters.setUptimeLookbackWindow(24)
+      await blockchainParameters.setUptimeLookbackWindow(50)
+      assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), 24)
+      await mineBlocks(EPOCH, web3)
+      assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), 50)
+    })
+
+    it('when doing 2 sets, should override the value for the next epoch', async () => {
+      // enter steady state
+      await blockchainParameters.setUptimeLookbackWindow(24)
+      await mineBlocks(EPOCH, web3)
+
+      await blockchainParameters.setUptimeLookbackWindow(lookbackWindow)
+      await blockchainParameters.setUptimeLookbackWindow(50)
+
+      // current epoch still initial value
+      assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), 24)
+
+      // after the epoch, find last value set
+      await mineBlocks(EPOCH, web3)
+      assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), 50)
+    })
+
+    it('should emit the corresponding event', async () => {
+      const resp = await blockchainParameters.setUptimeLookbackWindow(lookbackWindow)
+      assert.equal(resp.logs.length, 1)
+      const log = resp.logs[0]
+      assertContainSubset(log, {
+        event: 'UptimeLookbackWindowSet',
+        args: {
+          window: new BigNumber(lookbackWindow),
+        },
+      })
+    })
+
+    it('only owner should be able to set', async () => {
+      await assertRevert(
+        blockchainParameters.setUptimeLookbackWindow(lookbackWindow, {
+          from: accounts[1],
+        })
+      )
+    })
+
+    it('should fail when using value lower than safe minimum', async () => {
+      await assertRevert(
+        blockchainParameters.setUptimeLookbackWindow(11, {
+          from: accounts[1],
```

Comment:
This will revert just because it's not the owner

Reply:
dumb :roll_eyes: 
Thanks